pyveda/vedaset/stream/vedastream.py

```diff
@@ -110,12 +110,20 @@ def exhausted(self, val):
 
     @property
     def images(self):
-        _, imgs = zip(*self._vset._buf)
+        try:
+            _, imgs = zip(*self._vset._buf)
+        except ValueError:
+            time.sleep(.5)
+            return self.images
         return BufferedVariableArray(imgs)
```

Comment:
This never happens?

Reply:
I think ` _, imgs = zip(*self._vset._buf)` will always `ValueError` when initially called

Reply:
Yup, there's nothing to make sure there's actually a value in the buffer to return. So the simple solution was to pause for a bit to git the buffer a chance to fill. This should probably be smarter, for example by only retrying a couple of times.

Reply:
I can modify so it only re-tries like 3 times then push up the new code

Reply:
Sure, unless @rjpolackwich knows some way to get the status of the buffer so we can wait until it gets data loaded while also knowing if it's failing.

Reply:
There are ways to know if the buffers are loading that are already built in, eg `exhausted`. There are ways to get even more information about current state, but those are not yet built. In the meantime, I guess I don't see what's wrong with catching the `ValueError`, and re-raising it with a custom exception message?  Like I said have mentioned here, what if an error occurs upstream to the buffer fill? That includes server upstream, network upstream, and client code upstream, which is a bunch of places where things could fail.  

This is a buffer-based data accessor that pulls in the background in sync with the iteration interface on the vedastream api, so basically that means that data access via get and whatnot and all the methods around that need to respect the iteration that drives data fetching. Sleeping here is wasteful when our steamer is exhausted, for instance, or when the start hasn't been started. 

I would suggest either 1) not considering this a bug, and reraising the ValueError (or some custom error) with a helpful message (there's nothing wrong with errors being raised), or else, if we want to introduce a little latency for the buffer to fill, why not do it in one place, like when the thread is started, instead of on a data access method?

Reply:
I agree the recursion concerns here but regardless the original issue still needs to be solved in some way. @rjpolackwich what we want is to basically until the buffer is filled, right?

Reply:
Wait, why aren't we using the `_initialize_buffer` function? This function solves this problem, and blocks.

Reply:
So "like when the thread is started" makes the most sense for me. Prevent access until the buffer is filled, otherwise when the buffer is exhausted why would not just return None? Maybe not fully following

Reply:
@chelm Sure, nothing wrong at all with return None

Reply:
The point of this convo is that we need to address these kinds of concerns in the right places the right way, is all. Part of that is understanding how the thing works, using what we have available, and writing safe code when we need to

Reply:
I think the problem is up stream and needs to be resolved there. Agree, time.sleep... no bueno. But we need to figure out where upstream its failing to block.

Reply:
not sure anyone is arguing that.

Reply:
Like I said, this fix isn't wise - it's a placeholder that there's a problem and gives a hint towards why it's happening. `vs.train[0]` never worked for us, and we aren't familiar enough with the code to know the best fix. By getting it to at least work we were able to move on with more testing.

Reply:
Right, familiarity with the codebase is the crucial first step here. If all of us know what everything does, at least conceptually at the accessor interface level (ie, I don't expect co-devs to necessarily understand the fetching code, for instance), we can all contribute relevantly which will allow us to iterate quickly.